File: benew-services/src/main/java/com/narciarz/benew/models/dto/LoginResponseDto.java
package com.narciarz.benew.models.dto;

import com.narciarz.benew.models.UserRole;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

/**
 * DTO for successful login response.
 * 
 * <p>Returned by POST /auth/login on successful authentication.
 * Contains JWT token and user information for client-side role-based access control.</p>
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class LoginResponseDto {
    
    /**
     * JWT authentication token for subsequent API calls.
     * Include this in Authorization header as Bearer token.
     */
    private String token;
    
    /**
     * User ID.
     */
    private UUID userId;
    
    /**
     * User email.
     */
    private String email;
    
    /**
     * User role (ADMIN, MANAGER, USER) for client-side access control.
     */
    private UserRole role;
    
    /**
     * User first name.
     */
    private String firstName;
    
    /**
     * User last name.
     */
    private String lastName;
}

File: benew-frontend/src/models/login-response.dto.ts
import { UserRole } from './user-role';

/**
 * DTO for successful login response.
 * Returned by POST /auth/login on successful authentication.
 */
export interface LoginResponseDto {
  /**
   * JWT authentication token for subsequent API calls.
   * Include this in Authorization header as Bearer token.
   */
  token: string;

  /**
   * User ID (UUID format).
   */
  userId: string;

  /**
   * User email.
   */
  email: string;

  /**
   * User role (ADMIN, MANAGER, USER) for client-side access control.
   */
  role: UserRole;

  /**
   * User first name.
   */
  firstName: string;

  /**
   * User last name.
   */
  lastName: string;
}
